File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.java
class Solution {
    public List<List<Integer>> combinationSum2(int[] arr, int target) {
        List<List<Integer>> list=new ArrayList<>();
        List<Integer> temp=new ArrayList<>();
        Arrays.sort(arr);
        int n=arr.length;
        solver(0,arr,list,temp,target,n);
        return list;
    }
    private void solver(int ind,int []arr,List<List<Integer>> list,List<Integer> temp,int target,int n){
        if(target==0){
            if(list.contains(temp)){
                return;
            }
            list.add(new ArrayList<>(temp));
            return;
        }
        if(ind>=n){
            return ;
        }
        if(target>=arr[ind]){
            temp.add(arr[ind]);
            solver(ind+1,arr,list,temp,target-arr[ind],n);
            temp.remove(temp.size()-1);
        }
        solver(ind+1,arr,list,temp,target,n);
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        seen = [] 
        if len(s) != len(t): 
            return False 
        
        for char in s:

            seen.append(char)
        
        for char_t in t: 
            if char_t in seen: 
                seen.remove(char_t)
            else:
                return False
        
        return True
        